pycbc/io/hdf.py

```diff
@@ -986,6 +988,91 @@ def to_coinc_xml_object(self, file_name):
 
         ligolw_utils.write_filename(outdoc, file_name)
 
+    def to_coinc_hdf_object(self, file_name):
+        ofd = h5py.File(file_name,'w')
+
+        # Some fields are special cases:
+        logging.info("Outputting end times")
+        time = self.get_end_time()
+        ofd.create_dataset('end_time', data=time, dtype=np.float32)
+
+        logging.info("Getting IFARs and outputting as FARs")
+        far = 1. / self.get_coincfile_array('ifar')
+        ofd.create_dataset('false_alarm_rate', data=far, dtype=np.float32)
+
+        far_exc = 1. / self.get_coincfile_array('ifar_exc')
+        ofd.create_dataset('false_alarm_rate_exclusive', data=far_exc,
+                           dtype=np.float32)
+
+        logging.info("Outputting p values")
+        fap = self.get_coincfile_array('fap')
+        ofd.create_dataset('p_value', data=fap,
+                           dtype=np.float32)
+
+        fap_exc = self.get_coincfile_array('fap_exc')
+        ofd.create_dataset('p_value_exclusive', data=fap_exc,
+                           dtype=np.float32)
+
+        # Coinc fields
+        logging.info("Outputting coinc info")
+        for field in ['stat']:
+            logging.info(field)
+            vals = self.get_coincfile_array(field)
+            ofd.create_dataset(field, data=vals, dtype=np.float32)
+
+        # Bank fields
+        logging.info("Outputting associated bank values")
+        for field in ['mass1','mass2','spin1z','spin2z']:
+            logging.info(field)
+            vals = self.get_bankfile_array(field)
+            ofd.create_dataset(field, data=vals, dtype=np.float32)
+
+        mass1 = self.get_bankfile_array('mass1')
+        mass2 = self.get_bankfile_array('mass2')
+        mchirp, _ = mass1_mass2_to_mchirp_eta(mass1, mass2)
+
+        ofd.create_dataset('chirp_mass', data=mchirp, dtype=np.float32)
+
+        sngl_fields = [f for f in self.sngl_files[self.ifos[0]].columns
+                       if not ('template' in f)]
+        # Single-detector fields
+        logging.info("Outputting single-detector info")
+        for field in sngl_fields:
+            if field in ['gating','search','snr']: continue
```

Comment:
This bit is the main difference from the other PR, as we get all of the single-detector columns from the trigger_merge files and then filter out the ones that we don't want